Initiate GOI module `README` (#246)
AlexsLemonade/scpca-downstream-analyses: optional-goi-analysis/README.md

# Optional Genes of Interest (GOI) Analysis

> This directory includes a genes of interest analysis workflow that can help users evaluate values specific to a list of genes against the rest of a sample dataset.

Reviewer: ```suggestion
This directory includes a genes of interest analysis workflow to help users evaluate expression of a specific list of genes in a sample dataset.
```

**The genes of interest (GOI) analysis workflow cannot be implemented until after users have successfully run the main downstream analysis core workflow as described in this repository's main [README.md](../README.md) file.**

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Analysis overview](#analysis-overview)
- [Expected input](#expected-input)
- [Expected output](#expected-output)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Analysis overview

There are two main steps of this genes of interest analysis workflow:

> 1. **Genes of Interest Calculations**: Provided gene identifiers are first optionally mapped to a specified keytype.

Reviewer: I think it would make more sense to break these steps up into something that makes more sense to external users and not just the two steps that we do which is calculations and plots. What actually is happening in the script? 

Maybe something like these headers? 

1. Mapping gene identifiers 
3. Heirarchical clustering of dataset with genes of interest - this is what the heatmap represents 
4. Visualization of genes of interest expression (Sina, PCA, and UMAP)

If you are using conda, dependencies can be installed as [part of the initial setup](../README.md#snakemakeconda-installation).

## Expected input

To run this workflow, you will need to provide:

1. The RDS file containing the normalized [output `SingleCellExperiment` object](../README.md#expected-output) from the core dowstream analyses workflow.
2. A project metadata tab-separated value (TSV) file containing relevant information about your data necessary for processing, the same metadata file used in the core workflow should be used here (see more on this in the ["Metadata file format" section](../README.md#metadata-file-format) and an example of this metadata file [here](../project-metadata/example-library-metadata.tsv)).
3. The genes of interest list, stored as a tab-separated value (TSV) file.
This file should contain at least one column named `gene_id` with the relevant gene identifiers, and can optionally contain an additional column named `gene_set` that denotes the gene set that each gene identifier belongs to (see an example of this genes of interest file [here](../example-data/goi-lists/sample01_goi_list.tsv)).

## Expected output

> For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return five files in the same directory where the inputted RDS file is stored:

Reviewer: I think have a tree structure would be helpful here.

> 1. **Optionally**, a `_mapped_genes.tsv` file with mapped genes of interest if the `--perform_mapping` is `TRUE` upon running the workflow.

Reviewer: This isn't optional anymore based on the recent changes. This will always be there, but will only include a mapped column if `--perform_mapping` is `TRUE`. It will always contain a `sce_rownames_identifier` column that has been added.

# Optional Genes of Interest (GOI) Analysis

This directory includes a genes of interest analysis workflow to help users evaluate expression of a specific list of genes in a sample dataset.

**The genes of interest (GOI) analysis workflow cannot be implemented until after users have successfully run the main downstream analysis core workflow as described in this repository's main [README.md](../README.md) file.**

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Analysis overview](#analysis-overview)
- [Expected input](#expected-input)
- [Expected output](#expected-output)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->

## Analysis overview

There are three main steps of this genes of interest analysis workflow:

> 1. **Mapping gene identifiers**: Provided gene identifiers are first mapped to a specified keytype when `--perform_mapping` is `TRUE`.

Reviewer: I think generally we are going to need to have more explanation on the `keytype` and also what it means to map identifiers. I think we may want to have a separate section explaining mapping identifiers in more detail and that can be addressed in a separate issue, but for now something like this? 

```suggestion
1. **Mapping gene identifiers (Optional) **: If input gene identifiers do not match the gene identifiers used in the `SingleCellExperiment` object used as input (e.g., input genes are gene symbols while the `SingleCellExperiment` object contains Ensembl identifiers), then input gene identifiers must first be mapped to a specified type. 
```

2. **Heirarchical clustering of the dataset with genes of interest**: Hierarchical clustering is performed on the normalized data associated with the provided genes of interest.
These clustering results are used to generate a heatmap that displays genes on the x-axis, cells on the y-axis, and uses color to indicate gene expression scaled using a z-score.

> This heatmap can used to identify if any clear structure or groupings of cells with similar gene expression patterns are present.

Reviewer: ```suggestion
This heatmap can be used to identify if any clear structure or groupings of cells with similar gene expression patterns are present.
```

3. **Visualization of genes of interest expression** (Sina, PCA, and UMAP plots):

> The normalized and transformed expression for each of the provided genes of interest in comparison to the mean expression of all genes present in the dataset are plotted using `geom_sina()`.

Reviewer: ```suggestion
The normalized and transformed expression for each of the provided genes of interest is compared to the mean expression of all genes present in the dataset and shown using a sina plot.
```
In these docs, it's more important to talk about what is happening rather than what methods you are using. So instead of saying which function you used, you want to be clear on what is actually happening to the input data and what type of visualizations you are making.

> Following are UMAP and PCA plots where each dot represents a cell and the color indicates the individual gene of interest’s expression.

Reviewer: ```suggestion
UMAP and PCA plots are also provided, where each dot represents a cell and the color indicates the individual gene of interest’s expression.
```

**Note** that the same [software requirements for the core workflow](../README.md#3-additional-dependencies) are also required for this clustering workflow.
R 4.2 is required for running our pipeline, along with Bioconductor 3.15.
Package dependencies for the analysis workflows in this repository are managed using [`renv`](https://rstudio.github.io/renv/index.html), which must be installed locally prior to running the workflow.
If you are using conda, dependencies can be installed as [part of the initial setup](../README.md#snakemakeconda-installation).

## Expected input

To run this workflow, you will need to provide:

> 1. The RDS file containing the normalized [output `SingleCellExperiment` object](../README.md#expected-output) from the core dowstream analyses workflow.

Reviewer: ```suggestion
1. The RDS file containing the normalized [output `SingleCellExperiment` object](../README.md#expected-output) from the core dowstream analyses workflow.
This `SingleCellExperiment` object must contain a log-normalized counts matrix in an assay named `logcounts`.
```
Let's be specific of what is actually required here which is that the `logcounts` assay must be present.

2. A project metadata tab-separated value (TSV) file containing relevant information about your data necessary for processing, the same metadata file used in the core workflow should be used here (see more on this in the ["Metadata file format" section](../README.md#metadata-file-format) and an example of this metadata file [here](../project-metadata/example-library-metadata.tsv)).
3. The genes of interest list, stored as a tab-separated value (TSV) file.
This file should contain at least one column named `gene_id` with the relevant gene identifiers, and can optionally contain an additional column named `gene_set` that denotes the gene set that each gene identifier belongs to (see an example of this genes of interest file [here](../example-data/goi-lists/sample01_goi_list.tsv)).

## Expected output

For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return the following files in the specified output directory:

```
goi_stats
├── library01_mapped_genes.tsv
├── library01_normalized_zscores.mtx
├── library01_heatmap_annotation.rds

> └── library01_goi-report-template.html

Reviewer: I think this might change when we get to the snakefile, as it would make sense to move the rendered html file out of the `goi_stats` folder, but this is fine for now.

This `SingleCellExperiment` object must contain a log-normalized counts matrix in an assay named `logcounts`.
2. A project metadata tab-separated value (TSV) file containing relevant information about your data necessary for processing, the same metadata file used in the core workflow should be used here (see more on this in the ["Metadata file format" section](../README.md#metadata-file-format) and an example of this metadata file [here](../project-metadata/example-library-metadata.tsv)).
3. The genes of interest list, stored as a tab-separated value (TSV) file.
This file should contain at least one column named `gene_id` with the relevant gene identifiers, and can optionally contain an additional column named `gene_set` that denotes the gene set that each gene identifier belongs to (see an example of this genes of interest file [here](../example-data/goi-lists/sample01_goi_list.tsv)).

## Expected output

For each provided `SingleCellExperiment` RDS file and associated `library_id`, the workflow will return the following files in the specified output directory:

```
goi_stats
├── library01_mapped_genes.tsv
├── library01_normalized_zscores.mtx
├── library01_heatmap_annotation.rds

> └── library01_goi-report-template.html

Reviewer: ```suggestion
└── library01_goi-report.html
```
Just removing `template` from the name to match the description below for now.
